Allow HTML pass-through for autocomplete form type

Yippy opened the pull request:
<!-- THE PR TEMPLATE IS NOT AN OPTION. DO NOT DELETE IT, MAKE SURE YOU READ AND EDIT IT! -->

<!--
    Show us you choose the right branch.
    Different branches are used for different things :
    - 3.x is for everything backwards compatible, like patches, features and deprecation notices
    - master is for deprecation removals and other changes that cannot be done without a BC-break
    More details here: https://github.com/sonata-project/SonataAdminBundle/blob/3.x/CONTRIBUTING.md#the-base-branch
-->
I am targeting this branch, because "Fix XSS vulnerability in autocomplete" #4796 prevent passing HTML through label string, due to security reason. I have added an new option called 'safe_label' which follows KNP menu render wording. By default the safe label is false.

<!--
    Specify which issues will be fixed/closed.
    Remove it if this is not related.
-->

Closes #4815

## Changelog

<!-- MANDATORY
    Fill the changelog part inside the code block.
    Follow this schema: http://keepachangelog.com/
-->

<!-- REMOVE EMPTY SECTIONS -->
```markdown
### Added
- Added new 'safe_label' option to allow HTML pass-through
```

## To do

<!--
    If this is a work in progress, COMPLETE and ADD needed tasks.
    You can add as many tasks as you want.
    If some are not relevant, just REMOVE them.
-->

- [x] Update the documentation

greg0ire commented on src/Resources/views/Form/Type/sonata_type_model_autocomplete.html.twig:
```diff
@@ -148,7 +148,11 @@ file that was distributed with this source code.
             var templateResult = function (item) {
                 return {% block sonata_type_model_autocomplete_dropdown_item_format -%}
-                    jQuery('<div class="{{ dropdown_item_css_class }}">').text(item.label).prop('outerHTML')
+                    {% if safe_label is defined and not safe_label %}
```
`not safe_label|default(true)` would be the same, right?

greg0ire commented:
Please associate your poyu.co.uk email address with your Github account, or change the email in your commits to an address already associated with it.

greg0ire commented on docs/reference/form_types.rst:
```diff
@@ -357,6 +357,10 @@ The available options are:
 
+``safe_label``
+  defaults to ``false``. Set to ``true`` to enable the label to be displayed as Raw HTML,
+  which may cause XSS vulnerability.
```
pick either "XSS vulnerabities" or "an XSS vulnerability"

greg0ire commented on src/Resources/views/Form/Type/sonata_type_model_autocomplete.html.twig:
```diff
@@ -148,7 +148,11 @@ file that was distributed with this source code.
             var templateResult = function (item) {
                 return {% block sonata_type_model_autocomplete_dropdown_item_format -%}
-                    jQuery('<div class="{{ dropdown_item_css_class }}">').text(item.label).prop('outerHTML')
+                    {% if safe_label is defined and safe_label %}
```
so how about `safe_label|default(false)`?

Yippy replied:
That's a new way of writing it at first it does not look like the same outcome, but there are similar if function statement written as
`{%- if read_only is defined and read_only %}`
on line 15, 21, 77

I can amend but would it not be consistent?

greg0ire replied:
I think it's ok to amend, if we choose to be consistent, we will never improve anything. If you want to go the extra mile, change the other usages too.

OskarStark commented on docs/reference/form_types.rst:
```diff
@@ -357,6 +357,10 @@ The available options are:
 
+``safe_label``
+  defaults to ``false``. Set to ``true`` to enable the label to be displayed as Raw HTML,
+  which may cause XSS vulnerability
```
please add a dot at the end

Yippy replied:
already done with commit 8e89a36e8836e62b8e51f86a37683c75e4cfbcd2

OskarStark commented on docs/reference/form_types.rst:
```diff
@@ -357,6 +357,10 @@ The available options are:
   defaults to "". CSS class of dropdown item.
 
+``safe_label``
+  defaults to ``false``. Set to ``true`` to enable the label to be displayed as Raw HTML,
```
Raw -> raw

greg0ire commented:
Please squash some of these commits together. If you don't, we'll try to remember to do it for you but it's best if you save us this trouble.

How to do that?

1. `git rebase -i origin/3.x`, assuming `origin` is a git remote that points to this repository, and not your fork. If you're not sure what your remotes are, run `git remote -vvv`, there should be your fork and the holy/reference/base/origin/whatever-you-call-it repository.
2. A window will show up with many lines, replace `pick` with `fixup` on commit you want to fixup with the commit above.
3. Close your editor, git should do its magic, and you should end up with one commit
4. Use `git push --force` to overwrite what you already pushed. Don't forget the `--force` option otherwise git will try to merge both things together.

greg0ire commented:
What happened? Tell me if I can be of some help, and use `git reflog` if you're not sure what you did, it will show you step by step.

Yippy commented:
first time, should I just create a new PR as I got another copy
```
81a69ca8 HEAD@{0}: rebase -i (start): checkout hub/3.x
f09ad184 HEAD@{1}: rebase -i (finish): returning to refs/heads/3.x
f09ad184 HEAD@{2}: rebase -i (start): checkout hub/3.x
f09ad184 HEAD@{3}: rebase -i (finish): returning to refs/heads/3.x
f09ad184 HEAD@{4}: rebase -i (start): checkout origin/3.x
f09ad184 HEAD@{5}: commit: Lowercase correction
43ba0f74 HEAD@{6}: commit: Use default filters https://twig.symfony.com/doc/2.x/filters/default.html, when checking undefined variable and value.
c4aa39e2 HEAD@{7}: commit: Reworded description
8e89a36e HEAD@{8}: commit: Added fullstop to description
0d329838 HEAD@{9}: commit: Updated form_types.rst documentation to include new option
36b46268 HEAD@{10}: commit: Ensure non defined 'safe_label' does not bypass XSS security
df5c04bf HEAD@{11}: commit: Allow HTML pass-through for autocomplete form type
```

greg0ire commented:
You can reopen this one and force push whatever you want over it

Yippy commented:
I don't know how to re-open this PR, I've followed this tutorial [https://gist.github.com/robertpainsi/2c42c15f1ce6dab03a0675348edd4e2c](url)

I've manage to re-instate the 7 commits back onto my fork, should just re-create a new PR

greg0ire commented:
No problem ^^
